Add functions to generate random variables from various distributions

Jeff-Clark opened the pull request:
@moebio please review

Added a bunch of functions I tried out while generating simulated data for a recent project. The base functions I chose to place in NumberOperators. A more natural place might be NumberGenerators but we don't currently have such a class.

I also added two functions to NumberListGenerators that use the most flexible of the random generators to make it easy to generate NumberLists with some interesting properties.

Added a repeatable 'random' function to NumberOperators based on our existing _Alea code. The new random distribution functions make use of it. This is to solve a problem I noticed in simulated data project where I wanted repeatable data over a large number of different calls into the framework. It's awkward to manage by passing different seeds into the various methods that support it.

The idea is to set the seed once then any sequence of calls to the functions that use it can give repeatable results if you desire. The default is to seed with date/time so unless you specify otherwise the results are non-repeatable. There is a push/pop mechanism to make it easy to seamlessly support passing a seed into a particular NumberListGenerators like we do in some of our functions.

If we like this idea we could (in a future PR) replace other uses of Math.random() with this NumberOperators.random so, in theory, we could get totally repeatable randomness in every function by setting one seed. I checked and was surprised to see that the Math.random() is about 50% slower than this _Alea-based random so performance would be better as well.

Some distributions generated with NumberListGenerators.createRandomBetaPERTDistribution:
![screen shot 2015-11-01 at 4 21 25 pm](https://cloud.githubusercontent.com/assets/1721620/10871445/37fcc606-80b5-11e5-99a7-57c85911b479.png)
![screen shot 2015-11-01 at 4 22 05 pm](https://cloud.githubusercontent.com/assets/1721620/10871448/3d9eb4ac-80b5-11e5-9f9d-0b8018effd0c.png)
![screen shot 2015-11-01 at 4 23 18 pm](https://cloud.githubusercontent.com/assets/1721620/10871449/42c69abc-80b5-11e5-9a79-f7e4755e1e87.png)
![screen shot 2015-11-01 at 4 24 13 pm](https://cloud.githubusercontent.com/assets/1721620/10871450/47f0dee4-80b5-11e5-83aa-7ae5172948d2.png)

And one bimodal:
![screen shot 2015-11-01 at 4 24 56 pm](https://cloud.githubusercontent.com/assets/1721620/10871451/506f3bd8-80b5-11e5-83be-a7aa5945040e.png)

Jeff-Clark commented:
This project might be useful for testing /jeff/test/distributions

moebio commented:
+1

Great work.

Coincidentally I recently added this function: 


/**
 * generates a numberList with random numbers in a normal distribution
 * @param  {Number} nValues size of generated numberList
 * @param  {Number} mean
 * @param  {Number} standardDeviation
 * @return {NumberList}
 * tags:random,generator,statistics
 */
NumberListGenerators.createRandomNormalDistribution 


(line 72) Is this redundant? Is the name, or the names of parameters inconsistent with your functions?

Jeff-Clark commented:
Yes, I saw NumberListGenerators.createRandomNormalDistribution. I just checked and found it was giving results with  the wrong standard deviation. It was off by a factor of 1/sqrt(18). I think related to the fact you used 6 added Random() numbers.

I changed to use my NumberOperators.normal method which gives correct results and is about 30% faster as well.

I also changed NumberOperators.normal to use your more friendly terms mean/standardDeviation rather than mu/sigma
